test/scripts/jenkins_xpack.sh

```diff
@@ -4,33 +4,48 @@ set -e
 
 export TEST_BROWSER_HEADLESS=1
 
-echo " -> Running mocha tests"
-cd "$XPACK_DIR"
-checks-reporter-with-killswitch "X-Pack Karma Tests" yarn test:browser
-echo ""
-echo ""
-
-echo " -> Running jest tests"
-cd "$XPACK_DIR"
-checks-reporter-with-killswitch "X-Pack Jest" node scripts/jest --ci --verbose
-echo ""
-echo ""
-
-echo " -> Running SIEM cyclic dependency test"
-cd "$XPACK_DIR"
-checks-reporter-with-killswitch "X-Pack SIEM cyclic dependency test" node legacy/plugins/siem/scripts/check_circular_deps
-echo ""
-echo ""
-
-# FAILING: https://github.com/elastic/kibana/issues/44250
-# echo " -> Running jest contracts tests"
-# cd "$XPACK_DIR"
-# SLAPSHOT_ONLINE=true CONTRACT_ONLINE=true node scripts/jest_contract.js --ci --verbose
-# echo ""
-# echo ""
-
-# echo " -> Running jest integration tests"
-# cd "$XPACK_DIR"
-# node scripts/jest_integration --ci --verbose
-# echo ""
-# echo ""
+if [[ -z "$CODE_COVERAGE" ]] ; then
+  echo " -> Running mocha tests"
+  cd "$XPACK_DIR"
+  checks-reporter-with-killswitch "X-Pack Karma Tests" yarn test:browser
+  echo ""
+  echo ""
+    
+  echo " -> Running jest tests"
+  cd "$XPACK_DIR"
+  checks-reporter-with-killswitch "X-Pack Jest" node scripts/jest --ci --verbose
+  echo ""
+  echo ""
+    
+  echo " -> Running SIEM cyclic dependency test"
+  cd "$XPACK_DIR"
+  checks-reporter-with-killswitch "X-Pack SIEM cyclic dependency test" node legacy/plugins/siem/scripts/check_circular_deps
+  echo ""
+  echo ""
+    
+  # FAILING: https://github.com/elastic/kibana/issues/44250
+  # echo " -> Running jest contracts tests"
+  # cd "$XPACK_DIR"
+  # SLAPSHOT_ONLINE=true CONTRACT_ONLINE=true node scripts/jest_contract.js --ci --verbose
+  # echo ""
+  # echo ""
+    
+  # echo " -> Running jest integration tests"
+  # cd "$XPACK_DIR"
+  # node scripts/jest_integration --ci --verbose
+  # echo ""
+  # echo ""
+else
+  echo " -> Running jest tests with coverage"
+  cd "$XPACK_DIR"
+  # build runtime for canvas
+  echo "NODE_ENV=$NODE_ENV"
+  node ./legacy/plugins/canvas/scripts/shareable_runtime
+  node scripts/jest --ci --verbose --coverage
+  # rename file in order to be unique one
+  test -f ../target/kibana-coverage/jest/coverage-final.json \
```

Comment:
This `test` will fail the build if the file isn't there (because it returns a non-zero exit code)

Also, you still might have the problem I mentioned before:

If `node scripts/jest --ci --verbose --coverage` has a test that fails, this script will exit before the rename happens. So, later on, when you expect the file to be called `xpack-coverage-final.json`, it will be called `coverage-final.json`. Is this a problem?

Reply:
Well, if any test will fail `coverage-final.json` won't be generated. And actually we agreed not to proceed with coverage merging if any tests failed.

So exiting before renaming is fine, build should be red and next step (merge) is not starting. If the tests pass, json file is expected to be generated and I rename it because the json with the same name will be generated for oss jest tests.
During merge step I extract both oss and xpack in the same folder for merging, names should be unique to avoid overwriting.
What do you think is the best solution with context I provided?

Reply:
I just asked @LeeDr and he feels we should not push the coverage data if there's a failure as that will skew the coverage numbers.

Reply:
All good, just wanted to make sure the file not getting renamed wasn't going to cause a downstream issue!